src/classes/CON_PrimaryAffls_TEST.cls

```diff
@@ -768,51 +768,492 @@ private with sharing class CON_PrimaryAffls_TEST {
     **************************************************************************************************************************/
 
+
+    /**************************************************************************************************************************************
+    * @description Helper method to generate error message associated with a mapping to a field that does not lookup to Account.
+    * @param  affiliationMapping An affiliation mapping with an invalid lookup field mapping.
+    * @return The expected error message associated with the generated model.
+    **************************************************************************************************************************************/
+    private static String getExpectedInvalidFieldMappingErrorMessage(Affl_Mappings__c affiliationMapping){
+        List<String> params = new List<String>{
+            Label.stgColContactPrimaryAfflField,
+            affiliationMapping.Primary_Affl_Field__c
+        };
+
+        return String.format(label.invalidContactFieldInAffiliationMapping, params);
+    }
+
+    /**************************************************************************************************************************************
+    * @description Helper method to generate error message associated with a mapping to an invalid record type.
+    * @param  affiliationMapping An affiliation mapping with an invalid record type mapping.
+    * @return The expected error message associated with the generated model.
+    **************************************************************************************************************************************/
+    private static String getExpectedInvalidRTMappingErrorMessage(Affl_Mappings__c affiliationMapping){
+        List<String> params = new List<String>{
+            Label.stgColAccountRecordType,
+            affiliationMapping.Account_Record_Type__c
+        };
+
+        return String.format(Label.invalidRecordTypeInAffiliationMapping, params);
+    }
+
+    /*************************************************************************************************************
+    * @description Test case to verify that inserting a contact with a primary affiliation field populated with record type
+    * validation enabled, a valid Contact field mapping and invalid Record Type mapping throws an error.
+    **************************************************************************************************************/
+    @isTest 
+    private static void insertContactAfflInvalidRecordTypeError(){
+        List<Contact> testContactsList = UTIL_UnitTestData_TEST.getMultipleTestContacts(100);
+
+        UTIL_CustomSettings_API.getSettingsForTests(new Hierarchy_Settings__c(Affiliation_Record_Type_Enforced__c = true));
+
+        String namespace = UTIL_Namespace.getNamespace();
+        String prefix = (String.isNotBlank(namespace)) ? namespace + '__' : '';
```

Comment:
In these situations, we can use `UTIL_Namespace.StrTokenNSPrefix(String)` to namespace prefix an SObject or field. I prefer using this as it leverages the caching and other handling in UTIL_Namespace. If there is a readability issue, then local variables might assist. This isn't a required change here, but something I'd prefer for us to do in the future since we already have the utility in place to handle it.